src/krb5/_change_password.pyx

```diff
@@ -0,0 +1,217 @@
+# Support for Microsoft set/change password was added in MIT 1.7
+
+import collections
+import typing
+
+from krb5._exceptions import Krb5Error
+
+from krb5._ccache cimport CCache
+from krb5._context cimport Context
+from krb5._creds cimport Creds
+from krb5._krb5_types cimport *
+from krb5._principal cimport Principal
+
+
+cdef extern from "python_krb5.h":
+    krb5_error_code krb5_set_password(
+        krb5_context context,
+        krb5_creds *creds,
+        const char *newpw,
+        krb5_principal change_password_for,
+        int *result_code,
+        krb5_data *result_code_string,
+        krb5_data *result_string
+    ) nogil
+
+    krb5_error_code krb5_set_password_using_ccache(
+        krb5_context context,
+        krb5_ccache ccache,
+        const char *newpw,
+        krb5_principal change_password_for,
+        int *result_code,
+        krb5_data *result_code_string,
+        krb5_data *result_string
+    ) nogil
+
+    krb5_error_code krb5_change_password(
+        krb5_context context,
+        krb5_creds *creds,
+        const char *newpw,
+        int *result_code,
+        krb5_data *result_code_string,
+        krb5_data *result_string
+    ) nogil
+
+def set_password(
+    Context context not None,
+    Creds creds not None,
+    const unsigned char[:] newpw not None,
+    change_password_for: typing.Optional[Principal] = None,
+) -> typing.Tuple[int, bytes, bytes]:
```

Comment:
This applies to the other APIs being added but I try and use a named tuple as the return result. This allows people to use it like a tuple or a more descriptive object with attributes. We can also document the meaning of each fields in the tuple definition rather than each method docstring.

While I don't have a named tuple example for a function, they are used as property return results in `_creds` and `_creds_mit`.

Reply:
`SetPasswordResult` is it then